daemon/api_sideload_n_try.go

```diff
@@ -270,6 +270,57 @@ func sideloadManySnaps(st *state.State, snapFiles []*uploadedSnap, flags sideloa
 	return chg, nil
 }
 
+var readComponentInfoFromCont = readComponentInfoFromContImpl
+
+func readComponentInfoFromContImpl(tempPath string) (*snap.ComponentInfo, error) {
+	compf, err := snapfile.Open(tempPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open container: %w", err)
+	}
+	return snap.ReadComponentInfoFromContainer(compf)
+}
+
+// readComponentInfo reads ComponentInfo from a snap component file and the
+// snap.Info of the matching installed snap. If instanceName is not empty, it
+// is used to find the right instance, otherwise the SnapName from the
+// component is used.
+func readComponentInfo(st *state.State, tempPath, instanceName string, flags sideloadFlags) (*snap.ComponentInfo, *snap.Info, *apiError) {
+	if !flags.dangerousOK {
+		// TODO read assertions for components
+		return nil, nil, BadRequest("only local components can be installed at the moment")
```

Comment:
this error doesn't really explain that dangerous needs to be provided?

"only unasserted installation of local component with --dangerous is supported at the moment"?